astroid/brain/brain_namedtuple_enum.py

```diff
@@ -644,14 +636,7 @@ def _get_namedtuple_fields(node: nodes.Call) -> str:
 
 def _is_enum_subclass(cls: astroid.ClassDef) -> bool:
     """Return whether cls is a subclass of an Enum."""
-    try:
-        return any(
-            klass.name in ENUM_BASE_NAMES
-            and getattr(klass.root(), "name", None) == "enum"
-            for klass in cls.mro()
```

Comment:
`cls.mro()` wasn't returning the `enum` ancestors in the case of the example on the original issue; hence the class never made became inferred.

Reply:
Looks like a slight performance improvement, too, based on the small sample size of profiling astroid.